### ISSUE TITLE
define a relation config struct that maps each coding guideline with one or more check identifiers

### ISSUE THREAD

**Issue by J-ZhengLi** [open-rust-initiative/eunomia#21]
For example:

```json
"guidelines": [
    "id": "xxx",
    "name": "xxxxx",
    "clippy": [
        {
            "level": "warn",
            "lint": "xxxxxx"
        }
    ]
]
```

meaning that a guideline item with `id`, and `name` could be checked by enabling a warning level clippy lint called `xxxxxx`

**Comment by J-ZhengLi** [comment]
claim

**Comment by J-ZhengLi** [comment]
closed via: #23
